wiki/Ranking_criteria/osu!mania/en.md

```diff
@@ -58,13 +63,14 @@ Overall rules and guidelines apply to every kind of osu!mania difficulty. Rhythm
 - **Two notes cannot be placed in a single column at the same timestamp.**
 - **No column can be left empty.** If you want to use fewer keys, change the number of keys in the Song Setup.
 - **There must not be more than 6 notes pressed at any given moment in Insane or lower difficulties.** Using more than 6 notes must also follow a reasonable spread to the next lower difficulty. This rule does not apply to ends of long notes, as they are released, not pressed.
-- **Beatmaps can only use 4 to 10 keys.** Anything else is not supported for the Ranked section.
-- **If multiple key modes are present in a single beatmap, the key mode must be denoted in all difficulty names. Otherwise, the key mode must not be denoted.**
-- **Beatmaps using the special N+1 style of mapping must be denoted as such under the `Advanced` tab of song setup in the editor and "N+1" must be added to the tags. For beatmaps containing multiple key modes, difficulties using this style must be denoted using N+1 notation (e.g. using 7K+1 instead of 8K on difficulty names).**
+- **Beatmaps can only use 4–10, 12, 14, 16, or 18 keys.**
+- **Each difficulty's name must denote its [playstyle](#common-terms), but only if its key count is above 10 or multiple playstyles are present in the beatmap.**
```

Comment:
Few things. This should be reworded in a way that...
- Any Non-Standard playstyle
- Multiple key modes in a beatmapset
- Different playstyles of one same key mode

...need to add their key mode/playstyle in their difficulty name.

Reply:
- Your first line to state "any non-standard playstyle" requiring playstyle identifier on difficulty names directly conflicts with past RC, where "every single keycount/playstyle mapset" must not include playstyle identifier on difficulty names.
- Multiple key modes already covered on the new clause, on the condition "multiple playstyles are present in the beatmap."
- Different playstyles in one mode also covered with this PR. This to avoid circumvention of ranking 8K maps with mixed 8K and 7K+1 spread.

As for why it have extra requirement for 12K+ maps, it's because there are many playstyles can be derived, but only the mapper knows how to actually play these maps. It will be difficult to impossible to play if this intended playstyle isn't followed.

Reply:
- The first point is something that was internally discussed. While the previous RC did not account for it, it is *now* been taking into consideration. A user is not going to know whether a beatmap with only one keymode being used in a non-standard playstyle unless they play the map. This is not intuitive, and adding the playstyle in such a case would be helpful.
- No. With the way playstyles are being treated differently from key counts, that is not the case. A multi-key mode set is not a multi playstyle set following the proposals' own definitions. Either this clause should disambiguate the terms, or they should be better clarified in the glossary. Or both, as proposed in the Review.
- Yes the current one does it. But that is the only thing it does.

I am not getting into 12+ playstyles. Simply commenting that the current wording is not clear for two of the three bulletpoints mentioned.

Reply:
- Not sure for the first point, but that does make sense. Does that mean for the future we should explicitly mention the playstyle?

I think you're right. We might need to put non-standards (mostly N+1s) descriptors.

Reply:
For now, we have to follow what we have. We can't enforce rules that haven't been officially merged in the RC afterall. Once that changes then we can start applying it moving on.

Reply:
> A multi-key mode set is not a multi playstyle set following the proposals' own definitions.

I don't really see how you came to this conclusion. there's no such thing as a playstyle that can span multiple key modes, based on the definition in this PR

> This should be reworded in a way that [...] Any Non-Standard playstyle [...] need to add their key mode/playstyle in their difficulty name.

this was intentionally avoided so that no rules change for key modes below 12 -- if it should change, probably do it in separate pr

Reply:
It would need to be discussed yes, but I personally don't see a reason why a playstyle lower than 10k should not be mentioned in the difficulty name just like 10k+ modes do.

While this is necessary for those modes, it still is something that lower keymodes can benefit from specifying as, again, players won't be scavenging through the tags to decypher what playstyle they're supposed to adapt to. Having it in the difficulty name is a surefire way for it to be understood at a first glance both from the website and client alike.

Reply:
It's almost always assumed that 4-10K uses standard playstyle. Even if the players played the wrong way, it's not as severe as 12K+ maps. For visibility of playstyles, I believe it will be seen on the map's tags (and maybe additionally description). For tags it will become more accessible on lazer.

It's true that we should discuss this in separate PR, because this PR's scope is on officially acknowledging 12K+ maps as rankable keycount/playstyles, not whether 4-10K requires identifier or not.

Reply:
> It would need to be discussed yes, but I personally don't see a reason why a playstyle lower than 10k should not be mentioned in the difficulty name just like 10k+ modes do.
> 
> While this is necessary for those modes, it still is something that lower keymodes can benefit from specifying as, again, players won't be scavenging through the tags to decypher what playstyle they're supposed to adapt to. Having it in the difficulty name is a surefire way for it to be understood at a first glance both from the website and client alike.

I agree with all of this, this PR is just not the place for it

as for the rest of your initial comment, those cases are already covered.

Reply:
marking as resolved but someone should mention this in the rc rewrite server thing.